library/cohesity_policy.py

```diff
@@ -0,0 +1,378 @@
+    '''
+    function to delete the protection policy
+    :param module: object that holds parameters passed to the module
+    :param policy_id: protection policy id
+    :return:
+    '''
+    try:
+        cohesity_client.protection_policies.delete_protection_policy(id=policy_id)
+    except APIException as ex:
+        raise__cohesity_exception__handler(
+            str(json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def get_cohesity_client(module):
+    '''
+    function to get cohesity cohesity client
+    :param module: object that holds parameters passed to the module
+    :return:
+    '''
+    try:
+        cluster_vip = module.params.get('cluster')
+        username = module.params.get('username')
+        password = module.params.get('password')
+        domain = 'LOCAL'
+        if "/" in username:
+            user_domain = username.split("/")
+            username = user_domain[1]
+            domain = user_domain[0]
+
+        cohesity_client = CohesityClient(cluster_vip=cluster_vip,
+                                         username=username,
+                                         password=password,
+                                         domain=domain)
+        return cohesity_client
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def main():
+    # => Load the default arguments including those specific to the Cohesity protection policy.
+    argument_spec = cohesity_common_argument_spec()
+    argument_spec.update(
+        dict(
+            name=dict(type='str', required=True),
+            description=dict(type='str', default=''),
+            state=dict(choices=['present', 'absent'], default='present'),
+            days_to_retain=dict(type=int, default=90),
+            incremental_backup_schedule=dict(type=dict, required=True),
+            full_backup_schedule=dict(type=dict),
+            blackout_window=dict(type=list),
+            retries=dict(type=int, default=3),
+            retry_interval=dict(type=int, default=30),
+            bmr_backup_schedule=dict(type=dict),
+            log_backup_schedule=dict(type=dict),
+            extended_retention=dict(type=list),
+            archival_copy=dict(type=list)
+        )
+    )
+
+    # => Create a new module object
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True)
+    results = dict(
+        changed=False,
+        msg="Attempting to manage Cohesity protection policy",
+        state=module.params.get('state')
+    )
+
+    global cohesity_client
+    cohesity_client = get_cohesity_client(module)
+    policy_exists, policy_details = get_policy_details(module)
+
+    if module.check_mode:
+        check_mode_results = dict(
+            changed=False,
+            msg="Check Mode: Cohesity protection policy doesn't exist",
+            id=""
```

Comment:
why is id "" ?

Reply:
check_mode_results is a result template, we will populate the id and return if the policy exist else it will be an empty string. This template is consistent with all other modules, for example for protection sources
check_mode_results = dict(
            changed=False,
            msg="Check Mode: Cohesity Protection Source is not currently registered",
            id="")

for protection Job
check_mode_results = dict(
            changed=False,
            msg="Check Mode: Cohesity Protection Job is not currently registered",
            id="")

for restore vm
check_mode_results = dict(
            changed=False,
            msg="Check Mode: Cohesity Protection Restore Job is not currently registered",
            id=""
        )

Reply:
Ok , It could have been just ignored if its just empty string. Good programming practise, unless you are re-init the string variable.